multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/zeistmanga/ZeistManga.kt

```diff
@@ -149,31 +169,139 @@ abstract class ZeistManga(
         return GET(url.toString(), headers)
     }
 
-    override fun searchMangaSelector(): String = ".grid.gtc-f141a > div"
-    override fun searchMangaFromElement(element: Element): SManga {
-        return SManga.create().apply {
-            setUrlWithoutDomain(element.select(".block").attr("href"))
-            title = element.selectFirst(".clamp.toe.oh.block")!!.text().trim()
-            thumbnail_url = element.selectFirst("img")!!.attr("src")
-        }
-    }
+    override fun searchMangaParse(response: Response) = popularMangaParse(response)
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val url = "$baseUrl/search".toHttpUrl().newBuilder()
-            .addQueryParameter("q", query)
-            .build()
+        val startIndex = maxResults * (page - 1) + 1
+        val url = apiUrl()
+            .addQueryParameter("max-results", (maxResults + 1).toString())
+            .addQueryParameter("start-index", startIndex.toString())
 
-        return GET(url.toString(), headers)
-    }
+        if (query.isNotBlank()) {
+            url.addQueryParameter("q", query)
+                .build()
+
+            return GET(url.toString(), headers)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is StatusList -> {
+                    url.addPathSegment(filter.selected.value)
+                }
+
+                is TypeList -> {
+                    url.addPathSegment(filter.selected.value)
+                }
+
+                is LanguageList -> {
+                    url.addPathSegment(filter.selected.value)
+                }
+
+                is GenreList -> {
+                    filter.state.forEach { genre ->
+                        when (genre.state) {
+                            true -> url.addPathSegment(genre.value)
+                            false -> {}
+                        }
+                    }
+                }
+
+                else -> {}
+            }
+        }
 
-    override fun searchMangaNextPageSelector(): String? = null
+        return GET(url.build().toString(), headers)
```

Comment:
Extensions-lib 1.4 now allows you to return the `HttpUrl` directly.

```suggestion
        return GET(url.build(), headers)
```

Can replace this in other places as well.